torch/fx/experimental/symbolic_shapes.py

```diff
@@ -251,21 +252,44 @@ class FloorDiv(sympy.Function):
         nargs = (2,)
         precedence = 50  # precedence of mul  # noqa: F811
 
+        # Default return type for SymPy assumptions.
+        # https://docs.sympy.org/latest/guides/assumptions.html#implementing-assumptions-handlers
+        is_real = True
+
+        @property
+        def base(self):
+            return self.args[0]
+
+        @property
+        def divisor(self):
+            return self.args[1]
+
         def _sympystr(self, printer):
-            lhs = self.args[0]
-            rhs = self.args[1]
-            lhs_str = printer.parenthesize(lhs, self.precedence)
-            rhs_str = printer.parenthesize(rhs, self.precedence)
-            return f"{lhs_str}//{rhs_str}"
+            base = printer.parenthesize(self.base, self.precedence)
+            divisor = printer.parenthesize(self.divisor, self.precedence)
+            return f"{base}//{divisor}"
+
+        # SymPy assumptions based on argument types.
+        def _eval_is_real(self):
+            return fuzzy_or([self.base.is_real, self.divisor.is_real])
+
+        def _eval_is_integer(self):
+            return fuzzy_and([self.base.is_integer, self.divisor.is_integer])
 
+        # Automatic evaluation.
+        # https://docs.sympy.org/latest/guides/custom-functions.html#best-practices-for-eval
         @classmethod
         def eval(cls, base, divisor):
             if base == 0:
                 return sympy.Integer(0)
-            if divisor == 1:
+            if isinstance(base, sympy.Integer) and divisor == 1:
```

Comment:
Don't you want to apply this simplification for any integral expression base aka we should query the assumption to see if it's integral instead of an isinstance check which will only let through literals?